### ISSUE TITLE
random next track

### ISSUE THREAD

**Issue by jef-pearlman** [ezio416/tm-totd-grinder#37]
I'd love it if there was an option to make the next track be random (from the list of eligible tracks) rather than the first.  I don't necessarily care about completing things in order as long as I'm making progress, and especially with things like TOTD it can be nice to jump around to different places.

(I actually implemented the random selection locally, but I haven't implemented the configuration yet and I'm not really familiar enough with addon development to build it into a PR.  And of course mine isn't signed so I can't _really_ use it.)

Thanks for a great addon!

p.s. Not sure if this is the best place to submit ideas/feature requests; I have a few more so I'll see what happens with this one first.

**Comment by jef-pearlman** [comment]
In case it's handy, here's the git diff.  Super-basic:

```
diff --git a/src/Main.as b/src/Main.as
index 8d54776..925fc83 100644
--- a/src/Main.as
+++ b/src/Main.as
@@ -678,6 +678,13 @@ void SetNextMap() {
             @nextMap = map;
     }

+    // Test random map selection.
+    if (mapsRemaining.Length > 0) {
+       uint randomMap = Math::Rand(0, mapsRemaining.Length - 1);
+       trace("maps remaining: " + mapsRemaining.Length + ", random choice: " + randomMap);
+       @nextMap = mapsRemaining[randomMap];
+    }
+
     if (metTargetTotal == maps.Length)
         allTarget = true;
     else {
```

**Comment by ezio416** [comment]
Thanks for the suggestion, and this is indeed the best place to put one! There are some other issues with the plugin I need to resolve including one which prevents this feature, but it can be done

**Comment by jef-pearlman** [comment]
Great; thanks!  I'll post my other ideas in separate issues, and you can use them or not as you like.  Also curious what else needs tweaking—I was wondering if #10 was going to cause problems since I never traced the medal detection code, but in testing it did detect me medaling on the random, non-first map.  (I still think #10 would be great, though.)

**Comment by jef-pearlman** [comment]
Also, there are some side things to implement, like the toggle to enable this and making the "next track" button randomness-aware so you can pick a different random track.

**Comment by ezio416** [comment]
Yes #10 is the major stopping block there, but also a game update broke some pb detection stuff that needs to be addressed before much else, I will get around to it somewhat soon
